File: WebAPI/WebAPI.Models/Models/Team.cs
using System;

namespace WebAPI.Models.Models
{
    public class Team
    {
        public Guid TeamId { get; set; }
        
        public Guid? ProjectId { get; set; }
        
        public string TeamName { get; set; }
        
        public string Location { get; set; }

        public Guid CustomerId { get; set; }
        
        public virtual int MembersCount { get; set; }
    }
}

File: web/src/types/team.ts
import { IUser } from './user';
import { IWorkSpace } from './workspace';

interface IBaseTeam {
    teamId: string;
    teamName: string;
    creationDate?: Date;
    location: string;
}

export interface ITeam extends IBaseTeam {
    users: IUser[];
    projectId?: string;
    membersCount: number;
}

export interface ITeamSimpleModel extends IBaseTeam {}

export interface ITeamPage {
    workSpace: IWorkSpace;
    team: ITeam;
}
